src/RockPaperScissors/RpsServer/Controllers/RpsController.cs

```diff
@@ -0,0 +1,210 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using RpsServer.Models;
+
+namespace RpsServer.Controllers
+{
+    [Route("api/[controller]")]
+    public class RpsController : Controller
+    {
+        private readonly GameContext context;
+
+        public RpsController(GameContext context)
+        {
+            this.context = context;
+        }
+
+        #region APIs
+        [HttpGet]
+        public IActionResult CreateUser()
+        {
+            User newUser = new User();
+            this.context.Users.Add(newUser);
+            this.context.SaveChanges();
+            return new ObjectResult(newUser.Id);
+        }
+
+        [HttpGet("{playerId}")]
+        public IActionResult GetOrCreateGame(Guid playerId)
+        {
+            // Assumption: Existing game with this player doesn't exist
+            Game game = this.AddOrUpdateGame(playerId);
+            Player player = this.AddOrUpdatePlayer(playerId, game.Id);
+            this.context.SaveChanges();
+            return new OkResult();
+        }
+
+        [HttpGet("status/{playerId}")]
+        public IActionResult CheckStatus(Guid playerId)
+        {
+            Player player = this.context.Players.Find(playerId);
+            Game game = this.context.Games.Find(player.Game);
+            return this.GetGameStatus(game, playerId);
+        }
+
+        [HttpPost("{playerId}/{move}")]
+        public IActionResult MakeMove(Guid playerId, PlayerState move)
+        {
+            if (move == PlayerState.Waiting)
+            {
+                return new BadRequestResult();
+            }
+
+            Player player = this.context.Players.Find(playerId);
+            Game game = this.context.Games.Find(player.Game);
+
+            this.UpdatePlayerMove(game, player, move);
+
+            return this.GetGameStatus(game, playerId);
+        }
+        #endregion
+
+        #region GetOrCreateGame
```

Comment:
Yeah, that's actually something I wanted to talk about. With MVC, does code that is used by a Controller go into a subfolder within Controllers, or should this kind of code be found elsewhere?